modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/logsdb/qa/exceptions/NotEqualToMatcherException.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.datastreams.logsdb.qa.exceptions;
+
+/**
+ * Exception thrown when the contents of two objects do not match.
+ */
+public class NotEqualToMatcherException extends MatcherException {
+    public NotEqualToMatcherException(final String message) {
+        super(message);
```

Comment:
I would also include the base and test messages in the template and print them nicely here, to facilitate debugging.

Reply:
Yes I did this in an upcoming PR to make error messages easier to understand.